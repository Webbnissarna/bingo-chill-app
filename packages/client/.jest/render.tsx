import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
