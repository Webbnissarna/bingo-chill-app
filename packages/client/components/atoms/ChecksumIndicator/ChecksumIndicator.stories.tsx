import type { Story, StoryDefault } from "@ladle/react";
import ChecksumIndicator from ".";

interface StoryProps {
  actual: string;
  desired: string;
}

function StoryComponent({ actual, desired }: StoryProps) {
  return <ChecksumIndicator actual={actual} desired={desired} />;
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  actual: "123",
  desired: "abc",
};

Default.argTypes = {
  actual: {
    control: { type: "text" },
  },
  desired: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Checksum Indicator",
} satisfies StoryDefault;
