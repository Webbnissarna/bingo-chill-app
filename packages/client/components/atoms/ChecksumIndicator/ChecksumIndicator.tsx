import { CheckCircleFilled, CloseCircleFilled } from "@ant-design/icons";

export interface ChecksumIndicatorProps {
  actual?: string;
  desired?: string;
}

export default function ChecksumIndicator({
  actual,
  desired,
}: ChecksumIndicatorProps): JSX.Element {
  const haveBoth = !!actual && !!desired;
  const bothSame = actual === desired;

  return (
    <div className="font-text text-base text-snowStorm-0">
      <span>Checksum: </span>
      {haveBoth && bothSame && (
        <span className="text-aurora-3">
          {desired} <CheckCircleFilled />
        </span>
      )}
      {haveBoth && !bothSame && (
        <span className="text-aurora-0">
          {actual} (actual) vs {desired} <CloseCircleFilled />
        </span>
      )}
      {!haveBoth && <span className="opacity-30">(N/A)</span>}
    </div>
  );
}
