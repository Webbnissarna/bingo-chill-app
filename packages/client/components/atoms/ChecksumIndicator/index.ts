export { default } from "./ChecksumIndicator";
export * from "./ChecksumIndicator";
