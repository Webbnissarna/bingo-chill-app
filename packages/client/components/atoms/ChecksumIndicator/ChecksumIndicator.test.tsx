import { render } from "../../../.jest/render";
import ChecksumIndicator from ".";

describe("ChecksumIndicator", () => {
  it("renders", () => {
    const { asFragment } = render(<ChecksumIndicator actual="" desired="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
