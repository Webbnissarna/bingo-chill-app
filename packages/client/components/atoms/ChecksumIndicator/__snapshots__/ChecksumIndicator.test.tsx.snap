// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecksumIndicator renders 1`] = `
<DocumentFragment>
  <div
    class="font-text text-base text-snowStorm-0"
  >
    <span>
      Checksum: 
    </span>
    <span
      class="opacity-30"
    >
      (N/A)
    </span>
  </div>
</DocumentFragment>
`;
